examples/profile.js

```diff
@@ -123,4 +210,24 @@ app.MainController = function($http, $scope) {
 };
 
 
+/**
+ * @param {ol.Coordinate} coordinate The current pointer coordinate.
+ * @param {ol.geom.Geometry|undefined} geometry The geometry to snap to.
+ */
+app.MainController.prototype.snapToGeometry = function(coordinate, geometry) {
+  var closestPoint = geometry.getClosestPoint(coordinate);
+  // compute distance to line in pixels
+  var dx = closestPoint[0] - coordinate[0];
+  var dy = closestPoint[1] - coordinate[1];
+  var dist = Math.sqrt(dx * dx + dy * dy);
+  var pixelDist = dist / this['map'].getView().getResolution();
+
+  if (pixelDist < 8) {
+    this['profile'].highlight(closestPoint[2]);
+  } else {
+    this['profile'].highlight(-1);
+  }
+};
```

Comment:
Do we still need to export the profile and call the highlight or should it now use the directive highlight attribute?
If yes, then you can use `clearHighlight` instead of passing `-1`.